stock_inventory_discrepancy/models/stock_inventory.py

```diff
@@ -17,13 +17,13 @@ class StockInventory(models.Model):
         ('pending', 'Pending to Approve'),
         ('done', 'Validated')]
 
-    @api.one
     @api.depends('line_ids.product_qty', 'line_ids.theoretical_qty')
     def _compute_over_discrepancy_line_count(self):
-        lines = self.line_ids
-        self.over_discrepancy_line_count = sum(
-            d.discrepancy_percent > d.discrepancy_threshold
-            for d in lines)
+        for rec in self:
+            lines = rec.line_ids
+            rec.over_discrepancy_line_count = sum(
+                d.discrepancy_percent > d.discrepancy_threshold
```

Comment:
unless this record as different comparison black-magic methods this sum is going to fail -> summing booleans.
Am I wrong?

Reply:
I don't know, I only fixed a warning about using  `api.one` by adding a loop on `self` in the body of the method

Reply:
And I just checked: Python boolean can be promoted to integers. https://docs.python.org/2.4/lib/typesnumeric.html : "Booleans are a subtype of plain integers" and True + True == 2

Reply:
TIL tnx :smile:

Reply:
not that I think it is a cool idea to rely on this, from a maintainability pov, now that I think of it. 

@simahawk https://github.com/OCA/stock-logistics-warehouse/pull/458/commits/d78e0a8efdae4601fbeb07bcf2b183cabb947d68 is ok with you?

Reply:
:+1: